wd_helper.au3

```diff
@@ -683,96 +682,58 @@ EndFunc   ;==>_WD_FrameLeave
 
 	If @error Then
 		$iErr = $_WD_ERROR_GeneralError
 	Else
-		$vResult = __WD_FrameList_Internal($sSession, 'null', '', $bReturnHTML_Internally)
+		$vResult = __WD_FrameList_Internal($sSession, 'null', '')
 		$iErr = @error
 	EndIf
 
 	#Region - post processing
-	If $iErr = $_WD_ERROR_Success Then ; can occur when $sFilter is used
+	If $iErr = $_WD_ERROR_Success Then
 
 		; Strip last @CRLF
 		$vResult = StringTrimRight($vResult, 2)
 
 		; create array of frames from string returned from __WD_FrameList_Internal
 		_ArrayAdd($a_Result, $vResult)
 
-		; check the array backwards as elements may be removed
-		Local $bWasDeleted = False
-		For $i = UBound($a_Result) - 1 To 0 Step -1
-			$bWasDeleted = False
+		; check the results
+		For $i = 0 To UBound($a_Result) - 1
 			; find "calling frame" location - set $sStartLocation
 			If $a_Result[$i][$_WD_FRAMELIST_BodyID] = $sElement_CallingFrameBody Then $sStartLocation = $a_Result[$i][$_WD_FRAMELIST_Absolute]
 
 			; recalculate locations from absolute path on COL0 to relative path on COL1
 			$a_Result[$i][$_WD_FRAMELIST_Relative] = StringRegExpReplace($a_Result[$i][$_WD_FRAMELIST_Absolute], '\A' & $sStartLocation & '\/?', '')
-
-			; Decode HTML
-			If $bReturnHTML Or $sFilter <> '' Then
-				$a_Result[$i][$_WD_FRAMELIST_HTML] = BinaryToString($a_Result[$i][$_WD_FRAMELIST_HTML], $SB_UTF8)
-			EndIf
-
-			; apply filter - delete frames that HTML content string do not fits a given regular expression pattern
-			If $sFilter <> '' Then
-				; check $s_HTML content
-				If StringRegExp($a_Result[$i][$_WD_FRAMELIST_HTML], $sFilter, $STR_REGEXPMATCH) = 1 Then
-					; do nothing keeping current row of $a_Result[$i][....]
-				ElseIf @error = 2 Then ; @error StringRegExp : Bad pattern. @extended = offset of error in pattern
-					$iExt = @extended
-					$iErr = $_WD_ERROR_InvalidExpression
-					ExitLoop
-				Else ; StringRegExp = 0 no MATCH
-					_ArrayDelete($a_Result, $i)
-					$bWasDeleted = True
-				EndIf
-			EndIf
-
-			; cleanup HTML
-			If Not ($bWasDeleted Or $bReturnHTML) Then
-				$a_Result[$i][$_WD_FRAMELIST_HTML] = ''
-			EndIf
 		Next
 
-		; check if anything stay in the array to check the case when $sFilter was used
-		If UBound($a_Result) = 0 Then
-			$iErr = $_WD_ERROR_NotFound
-			$a_Result = ''
+		; select desired DataType for the $vResult - usually string is option for testing and asking support
+		If $bReturnAsArray Then
+			$vResult = $a_Result
 		Else
-			; select desired DataType for the $vResult
-			If $bReturnAsArray Or $bReturnHTML Then ; do not return huge amount of HTML data to string - only as array as they can contain restricted data
-				$vResult = $a_Result
-			Else ; string is option for testing and asking support thus should not return huge amount of HTML data to string as they can contain restricted data
-				$vResult = _ArrayToString($a_Result)
-				If @error Then $vResult = ''
-			EndIf
+			$vResult = _ArrayToString($a_Result)
+			If @error Then $vResult = ''
```

Comment:
Shouldn't $iErr and $sMessage get updated here since this affects the data returned if _ArrayToString fails?

Reply:
Please take a look:
[added $_WD_ERROR_RetValue](https://github.com/Danp2/au3WebDriver/pull/362/commits/00a0e77562061316eec4a5670595ad40bcaec3d1)
and
[header suplemented](https://github.com/Danp2/au3WebDriver/pull/362/commits/a95a827e941555d30e6e24885155e11e41932271)